artifactory/commands/transferfiles/longpropertycheck.go

```diff
@@ -0,0 +1,272 @@
+	cmdutils "github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/progressbar"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	servicesUtils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"golang.org/x/exp/slices"
+	"io"
+	"sync"
+
+	loguitils "github.com/jfrog/jfrog-cli-core/v2/utils/log"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"time"
+)
+
+const (
+	propertyAqlPaginationLimit = 1000000
+	maxThreadCapacity          = 5000000
+	threadCount                = 10
+	maxAllowedValLength        = 2400
+	longPropertyCheckName      = "Properties with value longer than 2.4K characters"
+)
+
+// Property - Represents a property of an item
+type Property struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// valueLength - Equals to the value length
+func (p *Property) valueLength() uint {
+	return uint(len(p.Value))
+}
+
+// FileWithLongProperty - Represent a failed instance; File with property that has failed the check (used for csv report format)
+type FileWithLongProperty struct {
+	// The file that contains the long property
+	api.FileRepresentation
+	// The length of the property's value
+	Length uint `json:"value-length,omitempty"`
+	// The property that failed the check
+	Property
+}
+
+type LongPropertyCheck struct {
+	producerConsumer parallel.Runner
+	filesChan        chan FileWithLongProperty
+	errorsQueue      *clientutils.ErrorsQueue
+}
+
+func NewLongPropertyCheck() *LongPropertyCheck {
+	return &LongPropertyCheck{}
+}
+func (lpc *LongPropertyCheck) Name() string {
+	return longPropertyCheckName
+}
+
+func (lpc *LongPropertyCheck) ExecuteCheck(args cmdutils.RunArguments) (passed bool, err error) {
+	// Init producer consumer
+	lpc.producerConsumer = parallel.NewRunner(threadCount, maxThreadCapacity, false)
+	lpc.filesChan = make(chan FileWithLongProperty, threadCount)
+	lpc.errorsQueue = clientutils.NewErrorsQueue(1)
+	var waitCollection sync.WaitGroup
+	var filesWithLongProperty []FileWithLongProperty
+	// Handle progress display
+	var progress *progressbar.TasksProgressBar
+	if args.ProgressMng != nil {
+		progress = args.ProgressMng.NewTasksProgressBar(0, progressbar.GREEN, "long property")
+		defer progress.GetBar().Abort(true)
+	}
+	// Create consumer routine to collect the files from the search tasks
+	waitCollection.Add(1)
+	go func() {
+		for current := range lpc.filesChan {
+			filesWithLongProperty = append(filesWithLongProperty, current)
+		}
+		waitCollection.Done()
+	}()
+	// Create producer routine to create search tasks for long properties in the server
+	go func() {
+		defer lpc.producerConsumer.Done()
+		lpc.longPropertiesTaskProducer(progress, args)
+	}()
+	// Run
+	lpc.producerConsumer.Run()
+	close(lpc.filesChan)
```

Comment:
can't use defer because we need to wait for all the routines to finish in order to get the results, one of them can finish only when the channel is closed, we don't have any error catching before that so we know we will get to this point .